File: typescript/project-euler-solutions/test/P0002.spec.ts
import {P0002} from '../src/P0002';
import {Common} from '../src/common';

describe('Helpers', () => {
  test('The 6th (0 based index) Fibonacci number" should "be 8', () => {
    const ans = Common.fibonacci(6);
    expect(ans).toBe(8);
  });

  test('The sum of the even-valued terms of the Fibonacci sequence below 4000000 should be 4613732', () => {
    const ans = P0002.sumEvenValuedFiboBelow(4000000);
    expect(ans).toBe(4613732);
  });
});

File: haskell/project-euler-solutions/test/P0002Spec.hs
module P0002Spec where

import P0002

import Test.Hspec (Spec, describe, context, it, shouldBe)

-- |Required for auto-discpvery
spec :: Spec
spec =
  describe "Project Euler Problem 2 Solution" $ do

    describe "fibonacci 6" $ do
      it "the 6th (0 index based) Fibonacci's number is 8" $ do
        show (fibonacci 6) `shouldBe` "8"

    describe "sumEvenFibonacciUntil 4000000" $ do
      it "the sum of the even-valued terms of Fibonacci below 4000000 is 4613732" $ do
        show (sumEvenFibonacciUntil 4000000) `shouldBe` "4613732"